src/uu/rm/src/rm.rs

```diff
@@ -437,8 +437,23 @@ fn remove_file(path: &Path, options: &Options) -> bool {
     false
 }
 
+fn prompt_dir(path: &Path, options: &Options) -> bool {
+    // If interactive is Never we never want to send prompts
+    if options.interactive == InteractiveMode::Never {
+        return true;
+    }
+
+    // We can't use metadata.permissions.readonly for directories because it only works on files
+    // So we have to handle whether a directory is writable manually
+    if let Ok(metadata) = fs::metadata(path) {
+        handle_writable_directory(path, options, &metadata)
+    } else {
+        true
+    }
+}
+
 #[allow(clippy::cognitive_complexity)]
```

Comment:
Maybe we can remove this now?

Reply:
No, there is still a warning from the `clippy::cognitive_complexity` lint :|

Reply:
:(